### ISSUE TITLE
Freemint 1.19

### ISSUE THREAD

**Issue by sagudev** [freemint/freemint#85]
When Freemint 1.19 will be released? Freemint 1.18 was released more than a year ago.

**Comment by ps68060** [comment]
Why not schedule an annual release and bump the version number accordingly ?  ;-)

**Comment by mikrosk** [comment]
I guess there's not many reasons why not to left. Personally I'd like to take a look at #73, #41 and #32 and if @DavidGZ could find some closure to #70, that would be a really good start to think about release.

Also, it depends how much time @atic-atac has for updating README and other bits and pieces needed for a release.

**Comment by th-otto** [comment]
I think lots of other files in doc/ also need to be updated before a release, a lot of information there is not valid any more.

**Comment by mikrosk** [comment]
Yes, the docs are a mess. See also https://github.com/freemint/freemint/issues/2. Personally it hurts me to see it in such state but then again, an Atari developer's time should be invested in a better activity than sorting and converting docs. :-/

**Comment by ps68060** [comment]
I am just bumping this suggestion of going for a release of 1-19. I know some of the issues above are still open but are they likely to be closed in the near future ?

https://github.com/freemint/freemint/issues/70 is closed

**Comment by paulwratt** [comment]
is it just the docs atm, I think it would be fine/good to move to a yearly release. why were there no point releases for 1.17 onwards? How come there is also no 1.16.3 on the release page, that the last good version before 1.17.0.

**Comment by ps68060** [comment]
Mint is as stable and reliable as always and there seems to be a bit of a lull in new developments.  A good time for a release ?  

1.19  or 1.20  (2020)  ?

**Comment by mikrosk** [comment]
Looking at the issues (even those I had mentioned earlier) I would like to see some of them fixed/sorted out. Also, Latz has been doing a wonderful job with documentation, I have a semi-final proposal pending in my mailbox.

Anyway, with all that Travis/Bintray stuff going on, why do you need it so urgently? New release will be current build + some README added.

**Comment by ps68060** [comment]
It's not urgent but it has been a few years since the last official release and I think it would be good to be able to announce a new version. 

It's a small thing but does give a sense of achievement and an opportunity for us users to say thanks.

**Comment by ps68060** [comment]
It's been 5 1/2  years since the last release.

https://github.com/freemint/freemint/releases

Any chance of creating a new one ?

**Comment by mikrosk** [comment]
Again, see https://github.com/freemint/freemint/issues/85#issuecomment-396264641 -- it's actually 9 years. ;-)

I'm more and more inclined to have FreeMiNT like a rolling updates kind of "distro". One thing is missing, though -- instead of having "releases" and "snapshots" we should have just latest release in the releases section updated after each commit.

So random people looking for a working release would find it immediately even on github.

Another step could be that some particular pre-release/snapshot could be marked as stable enough or with some interesting feature and mark that as a release (for instance right now we have a really nicely working USB thanks to @Perdrix24, @czietz and @DavidGZ, that is certainly a reason to encourage users to use it more.).

And yet another interesting point is the inclusion of #241 XaAES as an option.

**Comment by GokMasE** [comment]
> And yet another interesting point is the inclusion of #241 XaAES as an option.

Yes, I would definitely like to see XaAES-Ozk available as an official alternative.

I suppose it needs its own branch on Github to handle that properly?

**Comment by th-otto** [comment]
As already said in other places, that is IMHO not an option. That would essentially double the efforts in maintaining FreeMint, and lead to totally confusion for users, and also for us when reading bug-reports.

**Comment by mikrosk** [comment]
It requires a separate branch but that's not a big deal (it's been already done).

@th-otto as the classic in open source development says, it is the one who writes code who makes decision. :) If there are people willing to maintain it, report bugs and use it, I don't care.

For the time being I don't mind backporting bugfixes. You may have noticed that for instance @GokMasE reports bugs to both builds, so it's the other way around, it helps me to realise whether it is a regression or missing feature altogether. Works for me.

**Comment by ps68060** [comment]
What are the latest thoughts on having a better official release cycle ?

**Comment by ps68060** [comment]
Bump:

https://github.com/freemint/freemint/releases

Last release was 2017.

Why not just do an annual release using the year number ?
1-26-0 ?

**Comment by th-otto** [comment]
What is so urgent about an "official" release? It will be obsoleted by the next snapshot when someone changes something. And it will cause trouble for people that still have hard-coded names like 1-19-cur in some of their scripts.

**Comment by mikrosk** [comment]
First, as mentioned before and then again before that, the last release was 2013. It's all in this thread.

Second, I don't see a point of changing the number just for the sake of looking nice (26 = 2026). What's wrong with current IDs? You can have either `1-19-cur` or `1-19-<commit id>`, both of them serving their right purpose.

The official release requires at least one important thing and that is thorough testing. I don't see army of people reporting bugs. And when some do, there's only VERY FEW of them who are willing to cooperate on narrowing the bug down, messing with their setups etc. So usually bugs are fixed when devs find them and surprise, surprise, there is very few of us left.

There's nothing wrong with current snapshots, at least I'm not aware of anything critical (well except the mysterious https://github.com/freemint/freemint/issues/308 but it seems nobody is using CT2 and FreeMiNT except me) so take the latest snapshot and promote it in your head as a release. Done.

However I'd like to do one thing mention in this thread: similar as @czietz does in e.g. https://github.com/czietz/picowifi, make the snapshots latest beta release on github so it is more visible for random newcomers.

**Comment by ps68060** [comment]
It makes Mint look like a dead project.

Anyone coming to the main site that sees a there hasn't been a release in years isn't likely to download and install it. They won't look any further and might never see the snapshots.

There isn't a huge user-base so there aren't going to be many bug reports. Few bug reports is a good reason for making a release as any. The snapshots are always stable, compliment to the developers.

For the developers, making a release puts a line in the sand, it will mark a stable release. That might enable some more experimental work in the snapshots, it will matter less if they are broken/buggy for a few days or weeks.

Every project has periodic releases. It does give a sense of achievement.

**Comment by ps68060** [comment]
Why would they be responsible ? It's a group project.  I don't see how it's any different to the current situation for the devs.

I think there is benefit in a periodic "release".  I don't see the point in endless snapshots that are actually being treated like releases.

I've installed about 200 releases and they are all 1-19-0 which makes no sense to me.
